Mixpanel/MPNotification.m

```diff
@@ -1,9 +1,51 @@
 #import "MPLogger.h"
 #import "MPNotification.h"
 #import "Mixpanel.h"
+#import "MPNetworkPrivate.h"
 
+NSString* const PROPERTY_EVAL_FUNC_NAME = @"evaluateFilters";
+NSString* const PROPERTY_FILTERS_JS_URL = @"http://cdn-dev.mxpnl.com/libs/mixpanel.property-filters.dev.js";
 @implementation MPNotification
 
+#if !TARGET_OS_WATCH
++ (JSValue*) propertyFilterFunc {
+    static JSValue* propertyFilterFunc = nil;
+    static JSContext* jsContext = nil;
+    
+    if (propertyFilterFunc != nil) {
+        return propertyFilterFunc;
+    }
+    
+    jsContext = [[JSContext alloc] init];
+    
+    NSURL *url = [NSURL URLWithString:PROPERTY_FILTERS_JS_URL];
+    [[[MPNetwork sharedURLSession] dataTaskWithURL:url completionHandler:^(NSData *responseData, NSURLResponse *urlResponse, NSError *error) {
+        if (error) {
```

Comment:
`dataTaskWithURL:url` is async, so `return propertyFilterFunc;` at the end of this method won't work. Since it's a tiny file to download and this will always run in a background thread, you can just use `NSData dataWithContentsOfURL...`. it's sync and much more straightforward. Once you get NSData, you can convert it to NSString.